### DISCUSSION TITLE
After building there are two shallow.mjs version in the build

### DISCUSSION THREAD

**Discussion by budarin** [pmndrs/zustand#2526] [General] [answered]
"zustand": "^4.5.2"

<img width="759" alt="image" src="https://github.com/pmndrs/zustand/assets/8055157/e018c700-38c8-4491-b30b-7cfa470edbe7">

the shallow.js version from react is not needed - it is already supplied with react.

**Comment by dai-shi** [comment, marked as answer]
Yes, we are aware of doubling the code in bundle.
It's fixed in `v5`. Please try v5-alpha.

**Comment by dbritto-dev** [comment]
@budarin there're two shallow files: shallow (vanilla) and shallow (react)

The shallow vanilla export shallow()
The shallow react export useShallow()
